groovy/gradle/src/org/netbeans/modules/gradle/spi/newproject/BaseGradleWizardIterator.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.gradle.spi.newproject;
+
+import org.netbeans.modules.gradle.api.GradleFiles;
+import org.netbeans.modules.gradle.newproject.ProjectAttriburesPanel;
+import java.awt.Component;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.swing.JComponent;
+import javax.swing.event.ChangeListener;
+import org.netbeans.api.progress.ProgressHandle;
+import org.netbeans.spi.project.ui.support.CommonProjectActions;
+import org.openide.WizardDescriptor;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+public abstract class BaseGradleWizardIterator implements WizardDescriptor.ProgressInstantiatingIterator<WizardDescriptor> {
```

Comment:
I'd move this class into implementation and just exposed the `abstract` method as an interface. Then there would be a factory method that turns the interface instance to `WizardDescriptor.ProgressInstantiatingIterator`. A kind of [don't mix API/SPI](http://wiki.apidesign.org/wiki/APIvsSPI) advice.